src/Timeline.js

```diff
@@ -74,24 +75,37 @@ export default class Timeline extends Axis {
               ? event.selection 
               : [event.selection[0] + buttonDomain, event.selection[1] - buttonDomain].sort((a, b) => a - b);
 
-      const domain = (this._brushing ? selection
-        : this._buttonBehaviorCurrent === "buttons" 
-          ? [event.sourceEvent.offsetX, event.sourceEvent.offsetX] : [event.selection[0], event.selection[0]])
-        .map(this._d3Scale.invert)
-        .map(Number);
-
-      const ticks = this._availableTicks.map(Number);
-
-      domain[0] = date(closest(domain[0], ticks));
-      domain[1] = date(closest(domain[1], ticks));
+      const domain = this._buttonBehaviorCurrent === "ticks"
+        ? (this._brushing 
+          ? event.selection 
+          : [event.sourceEvent.offsetX, event.sourceEvent.offsetX]).map(this._d3Scale.invert).map(Number) 
+        : (this._brushing 
+          ? selection
+          : [event.sourceEvent.offsetX, event.sourceEvent.offsetX]).map(Number);
+
+      const ticks = this._buttonBehaviorCurrent === "ticks"
+        ? this._availableTicks.map(Number) 
+        : this._d3Scale.range();
+
+      if (this._buttonBehaviorCurrent === "ticks") {
+        domain[0] = date(closest(domain[0], ticks));
+        domain[1] = date(closest(domain[1], ticks));
+      } 
+      else {
+        domain[0] = closest(domain[0], ticks);
+        domain[1] = closest(domain[1], ticks);
+      }
 
       const single = +domain[0] === +domain[1];
+      this._selection = this._buttonBehaviorCurrent === "ticks" 
+        ? single ? domain[0] : domain 
+        : single 
+          ? new Date(this._availableTicks[ticks.indexOf(domain[0])], 0, 1) 
+          : [new Date(this._availableTicks[ticks.indexOf(domain[0])], 0, 1), new Date(this._availableTicks[ticks.indexOf(domain[1])], 0, 1)];
```

Comment:
why are you using the JavaScript `Date` and not the d3plus `date` function? You can probably just use `date` and condense all of this code.